tools/test_crypto.py

```diff
@@ -0,0 +1,156 @@
+"""
+Copyright (c) 2018 ARM Limited
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import logging
+import os
+import sys
+import git
+
+from subprocess import check_call
+import argparse
+SCRIPT_DIRECTORY = os.path.dirname(os.path.realpath(__file__)) + "/"
+MBEDTLS_IMPORTER_DIRECTORY = SCRIPT_DIRECTORY + "../features/mbedtls/importer/"
+MBED_CRYPTO_DIRECTORY = MBEDTLS_IMPORTER_DIRECTORY + "/TARGET_IGNORE/mbedtls/crypto/"
+MBED_CRYPTO_TEST_DIRECTORY = MBED_CRYPTO_DIRECTORY + "/tests/"
+MBED_OS_TEST_DIRECTORY = SCRIPT_DIRECTORY + "../TESTS/"
+
+
+logger = logging.getLogger('PSA-CRYPTO Builder')
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description="This script build psa-crypto on-target tests."
+    )
+
+    parser.add_argument(
+        '-c',
+        '--clean',
+        action='store_true',
+        help="choose a specific test to compile"
+    )
+
+    parser.add_argument(
+        '-r',
+        '--run',
+        action='store_true',
+        help="Run psa crypto tests"
+    )
+
+    parser.add_argument(
+        '-b',
+        '--build',
+        action='store_true',
+        help="Build psa crypto tests"
+    )
+
+    parser.add_argument(
+        '-g',
+        '--generate',
```

Comment:
What's the difference between generate and deploy?

Reply:
@cmonr there are 2 different stages
1. deploy - import (trough the makefile in importer directory) the mbed-crypto library include the library tests
2. generate the tests (there are written in mbedtls format) and copy them into TESTS